worldedit-core/src/main/java/com/fastasyncworldedit/core/extension/factory/parser/mask/DefaultMaskParser.java

```diff
@@ -1,189 +1,218 @@
+import com.sk89q.worldedit.extension.input.NoMatchException;
+import com.sk89q.worldedit.extension.input.ParserContext;
+import com.sk89q.worldedit.extension.platform.Actor;
+import com.sk89q.worldedit.extent.Extent;
+import com.sk89q.worldedit.function.mask.Mask;
+import com.sk89q.worldedit.function.mask.MaskIntersection;
+import com.sk89q.worldedit.internal.command.CommandArgParser;
+import com.sk89q.worldedit.internal.util.Substring;
+import com.sk89q.worldedit.session.request.Request;
+import com.sk89q.worldedit.world.block.BaseBlock;
+import org.enginehub.piston.inject.MemoizingValueAccess;
+import org.enginehub.piston.suggestion.Suggestion;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class DefaultMaskParser extends FaweParser<Mask> {
+
+    public DefaultMaskParser(WorldEdit worldEdit) {
+        super(worldEdit);
+    }
+
+    @Override
+    public Mask parseFromInput(String input, ParserContext context) throws InputParseException {
+        if (input.isEmpty()) {
+            throw new SuggestInputParseException("No input provided", "", () -> Stream
+                    .of("#", ",", "&")
+                    .map(n -> n + ":")
+                    .collect(Collectors.toList())
+                    // TODO namespaces
+            );
+        }
+        Extent extent = context.getExtent();
+        if (extent == null) {
+            extent = Request.request().getExtent();
+        }
+        List<List<Mask>> masks = new ArrayList<>();
+        masks.add(new ArrayList<>());
+
+        final CommandLocals locals = new CommandLocals();
+        Actor actor = context.getActor();
+        if (actor != null) {
+            locals.put(Actor.class, actor);
+        }
+        try {
+            List<Map.Entry<ParseEntry, List<String>>> parsed = parse(input);
+            for (Map.Entry<ParseEntry, List<String>> entry : parsed) {
+                ParseEntry pe = entry.getKey();
+                final String command = pe.getInput();
+                String full = pe.getFull();
+                Mask mask = null;
+                if (command.isEmpty()) {
+                    mask = parseFromInput(StringMan.join(entry.getValue(), ','), context);
+                } else if (!worldEdit.getMaskFactory().containsAlias(command)) {
+                    // Legacy patterns
+                    char char0 = command.charAt(0);
+                    boolean charMask = input.length() > 1 && input.charAt(1) != '[';
+                    if (charMask && input.charAt(0) == '=') {
+                        return parseFromInput(char0 + "[" + input.substring(1) + "]", context);
+                    }
+                    if (char0 == '#') {
+                        throw new SuggestInputParseException(
+                                new NoMatchException("Unknown mask: " + full + ", See: //masks"),
```

Comment:
Deprecated in favor of components. Please create a translation message. Also, the latter is not a command. (Afaik) you can't list masks ingame at the moment. That used to be legacy-like behavior.